### ISSUE TITLE
Reproducible build fails

### ISSUE THREAD

**Issue by ianamunoz** [zcash/zcash#3915]
### Describe the issue
When pulling master and running the zcash-gitian build with a release tag, the build is not reproducible. Any two builds, on fresh machine and fresh install, do not produce the same checksums for the binary products created by the zcash-gitian build.

#### If so, please list the steps to reproduce below:
1. Run a gitian build from current master and use the tag `v2.0.4-rc1`
2. Check the outputs from multiple builds and verify they are not the same.

### Actual behaviour + errors
I ran individual gitian builds at commit hashes with a few changes from https://github.com/zcash/zcash-gitian/pull/54 and https://github.com/zcash/zcash/pull/3914.

I was able to reproduce builds at the first two commits since `v2.0.3`. At commit:
`6a20252282ec383c14878a9cca7693c1708cf8e2`

I was unable to create a reproducible build from then on.

### Logs

Below is the output from the gitian build that produced different hashes at the 3rd commit after v2.0.3 showing the change from reproducible hashsum to a not reproducible hash sum.

```
--
./data/gitian-2a/run.log:145:25f52d31fc0bda94140c8d9d944b232a3412519ac24feadfcf78229c03e41175  src/zcash-2.0.3.tar.gz
./data/gitian-2a/run.log-146-afc6b77175ff51471b292128622f59fbc7f0dcfc0348954636f30d2b0086ac35  zcash-2.0.3-linux64-debug.tar.gz
./data/gitian-2a/run.log-147-d61693f68ae442c4b1493ee02447be2d9b301a1206a2a5fcbb00d224566d40ee  zcash-2.0.3-linux64.tar.gz
./data/gitian-2a/run.log-148-5b9c0c701d167165632b66fa845d52bda3740efd8ea23883df48ea16b1774014  zcash-2.0.3-res.yml
./data/gitian-2a/run.log-149-Done.
--
./data/gitian-2b/run.log:145:25f52d31fc0bda94140c8d9d944b232a3412519ac24feadfcf78229c03e41175  src/zcash-2.0.3.tar.gz
./data/gitian-2b/run.log-146-afc6b77175ff51471b292128622f59fbc7f0dcfc0348954636f30d2b0086ac35  zcash-2.0.3-linux64-debug.tar.gz
./data/gitian-2b/run.log-147-d61693f68ae442c4b1493ee02447be2d9b301a1206a2a5fcbb00d224566d40ee  zcash-2.0.3-linux64.tar.gz
./data/gitian-2b/run.log-148-5b9c0c701d167165632b66fa845d52bda3740efd8ea23883df48ea16b1774014  zcash-2.0.3-res.yml
./data/gitian-2b/run.log-149-Done.
--
./data/gitian-3a/run.log:145:83310ffe2d47c8400fb9ff39aa4e82cdfa86640907ce4ea355c9ddaa6da2560a  src/zcash-2.0.3.tar.gz
./data/gitian-3a/run.log-146-c61a746f5a32a9564a39ffb41155b00ccc2035bceb6fc1db2ff8559add758e65  zcash-2.0.3-linux64-debug.tar.gz
./data/gitian-3a/run.log-147-71b4698623b6c6b37c8290147fe51c86028070efce4507065b8dbce28d975553  zcash-2.0.3-linux64.tar.gz
./data/gitian-3a/run.log-148-3384ba244fc4ba8029e2e24e14593c7d4eae8162785dfe4ebc02fe60cc4f0e27  zcash-2.0.3-res.yml
./data/gitian-3a/run.log-149-Done.
--
./data/gitian-3b/run.log:145:83310ffe2d47c8400fb9ff39aa4e82cdfa86640907ce4ea355c9ddaa6da2560a  src/zcash-2.0.3.tar.gz
./data/gitian-3b/run.log-146-6b31e71052b13e1d1b656d47d53be9dc63cff8f3a6d7350e473f880a64ddf4a2  zcash-2.0.3-linux64-debug.tar.gz
./data/gitian-3b/run.log-147-b4c941e2c3e81efd7988e7fc8f815a76b7e8e65203876cc3a36636986b8d585f  zcash-2.0.3-linux64.tar.gz
./data/gitian-3b/run.log-148-4b606359ab8f14d63ce3e48918107abf9c54735d7d9c8d865f6053c4f95558b4  zcash-2.0.3-res.yml
./data/gitian-3b/run.log-149-Done.
```

**Comment by garethtdavies** [comment]
My experience with differing hashes in 2.0.4-rc1 is logged in the community chat here: https://chat.zcashcommunity.com/channel/zcashd-team?msg=QbEux3XNZHdTqMNhk

**Comment by daira** [comment]
This should be fixed by #3919. Please verify @garethtdavies.

**Comment by ianamunoz** [comment]
It is fixed by a combination of either 

#3919 and https://github.com/zcash/zcash-gitian/pull/55.
or: 
#3919, this PR, and https://github.com/zcash/zcash-gitian/pull/54.

The minimum changes we can make for this release will not merge this PR, but do require an update to the outer VM in zcash-gitian.

We should merge this PR and make stretch the officially supported debian distribution in the next release and close #3651 also.

**Comment by bitcartel** [comment]
@ianamunoz When you say "this PR" in the comment above you mean https://github.com/zcash/zcash/pull/3914/ ?

**Comment by ianamunoz** [comment]
Whoops sorry, commenting on the issue not the PR. Yes @bitcartel  I have updated my comment for clarity.

**Comment by garethtdavies** [comment]
I just rebuilt the vagrant box and pushed 2.0.4 sigs and they seem to match the PR from @ianamunoz

**Comment by str4d** [comment]
Closing as resolved by #3919.
